From ed42b10b3150a815fc0ab921876f7fdea7cf9e4d Mon Sep 17 00:00:00 2001
From: Sebanc <sebanc.xda@gmail.com>
Date: Wed, 9 Oct 2019 21:49:52 +0200
Subject: [PATCH] Microsoft Surface Go devices: use different ath10k board
 firmware name

Microsoft Surface Go uses a specific board firmware which is not the one in the Linux kernel's firmware repository. Use a dedicated board firmware name in order to avoid the specific firmware being overwritten on each package update and maintain compatibility with other ath10k devices.
---
 drivers/net/wireless/ath/ath10k/core.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/wireless/ath/ath10k/core.c b/drivers/net/wireless/ath/ath10k/core.c
index 5210cff..d16f462 100644
--- a/drivers/net/wireless/ath/ath10k/core.c
+++ b/drivers/net/wireless/ath/ath10k/core.c
@@ -21,6 +21,7 @@
 #include <linux/dmi.h>
 #include <linux/ctype.h>
 #include <linux/pm_qos.h>
+#include <linux/surface_devices_dmi.h>
 #include <asm/byteorder.h>
 
 #include "core.h"
@@ -35,6 +36,8 @@
 #include "wmi-ops.h"
 #include "coredump.h"
 
+static const struct dmi_system_id devices[] = surface_go_devices;
+
 unsigned int ath10k_debug_mask;
 static unsigned int ath10k_cryptmode_param;
 static bool uart_print;
@@ -1896,6 +1899,9 @@ static int ath10k_init_hw_params(struct ath10k *ar)
 
 	ar->hw_params = *hw_params;
 
+	if (dmi_check_system(devices))
+		ar->hw_params.fw.board = "surface_go.bin";
+
 	ath10k_dbg(ar, ATH10K_DBG_BOOT, "Hardware name %s version 0x%x\n",
 		   ar->hw_params.name, ar->target_version);
 
-- 
2.20.1

